templates/templates.go

```diff
@@ -0,0 +1,511 @@
+					}
+					updated = fmt.Sprintf("Updated by <a href='/internal/user/%d'>%s</a> at %s<br>", u.UpdatedBy.UserID, name, u.UpdatedAt.String)
+				}
+			} else if u.UpdatedAt.Valid {
+				updated = fmt.Sprintf("Updated by UNKNOWN at %s<br>", u.UpdatedAt.String)
+			}
+			if u.DeletedBy.UserID != -1 {
+				if len(u.DeletedBy.Firstname) == 0 && len(u.DeletedBy.Nickname) == 0 && len(u.DeletedBy.Lastname) == 0 {
+					deleted = fmt.Sprintf("Deleted by UNKNOWN(%d) at %s<br>", u.DeletedBy.UserID, u.DeletedAt.String)
+				} else {
+					var name string
+					if u.DeletedBy.Firstname != u.DeletedBy.Nickname {
+						name = fmt.Sprintf("%s (%s) %s", u.DeletedBy.Firstname, u.DeletedBy.Nickname, u.DeletedBy.Lastname)
+					} else {
+						name = fmt.Sprintf("%s %s", u.DeletedBy.Firstname, u.DeletedBy.Lastname)
+					}
+					deleted = fmt.Sprintf("Deleted by <a href='/internal/user/%d'>%s</a> at %s<br>", u.DeletedBy.UserID, name, u.DeletedAt.String)
+				}
+			} else if u.DeletedAt.Valid {
+				deleted = fmt.Sprintf("Deleted by UNKNOWN at %s<br>", u.DeletedAt.String)
+			}
+			output.WriteString(fmt.Sprintf("<p>%sUser ID: %d<br>First name: %s<br>Nickname: %s<br>Last name: %s<br>Username: %s<br>Email: %s<br><br>Enabled: %s<br>Login type: %s<br>%sAvartar source: %s<br><br>%s%s%s%s%s%s</p>", deleted, u.UserID, u.Firstname, u.Nickname, u.Lastname, u.Username, u.Email, enabled, u.LoginType, ldap, avatar, perms.String(), roles.String(), lastLogin, created, updated, deletedBy))
+			return template.HTML(output.String())
+		},
+	}
+}
+
+func (t *Templater) permissionsParser(id int, p string) bool {
+	m := GetValidPermissions(permissions.Permissions(p))
+
+	u, err := t.User.GetUser(context.Background(), user.User{UserID: id})
+	if err != nil {
+		log.Printf("failed to get user for template(permissionParser): %+v", err)
+		return false
+	}
+
+	p1, err := t.User.GetPermissionsForUser(context.Background(), u)
+	if err != nil {
+		log.Printf("failed to get permission for template(permissionParser): %+v", err)
+		return false
+	}
+
+	for _, perm := range p1 {
+		if m[perm.Name] {
+			return true
+		}
+	}
+	return false
+}
+
+func GetValidPermissions(p permissions.Permissions) (m map[string]bool) {
+	m = make(map[string]bool)
+	if p.String() == permissions.MenuDisabled.String() {
+		m[p.String()] = true
+		return
+	}
+
+	m[p.String()] = true
+
+	switch p {
+	case permissions.ManageMembersAdmin:
+	case permissions.KeyCardAccess:
+	case permissions.BookingsAdmin:
+	case permissions.CalendarAdmin:
+	case permissions.CMSAdmin:
+	case permissions.Cobra:
+	case permissions.Director:
+	case permissions.EditNetStats:
+	case permissions.EmailEveryone:
+	case permissions.EquipmentAdmin:
+	case permissions.HiresAdmin:
+	case permissions.Inform:
+	case permissions.KeyListManage:
+	case permissions.MailingListAdmin:
+	case permissions.OfficerReports:
+	case permissions.Streamer:
+	case permissions.TechieTodo:
+	case permissions.VideoStats:
+	case permissions.WatchAdmin:
+		break
```

Comment:
These could be removed to make the `switch` shorter.

Reply:
This was a relic but was addressed I think in PR-10 and moved into `infrastructure/permission` and removed